src/Explorer/Explorer.tsx

```diff
@@ -422,7 +417,13 @@ export default class Explorer {
     connectionStatus.status = ConnectionStatusType.Connected;
     useNotebook.getState().setConnectionInfo(connectionStatus);
     useNotebook.getState().setNotebookServerInfo({
-      notebookServerEndpoint: userContext.features.notebookServerUrl || connectionInfo.data.notebookServerUrl,
+      notebookServerEndpoint:
+        (validateEndpoint(
+          userContext.features.notebookServerUrl,
+          allowedNotebookServerUrls.map((endpoint) => endpoint)
```

Comment:
Is there anyway we can avoid doing this conversion? Also we can use the spread operator to do this if we have to: `[...allowedNotebookServerUrls]`